internal/dao/pod.go

```diff
@@ -455,3 +457,63 @@ func in(ll []string, s string) bool {
 	}
 	return false
 }
+
+func (p *Pod) GetPodSpec(path string) (*v1.PodSpec, error) {
+	pod, err := p.GetInstance(path)
+	if err != nil {
+		return nil, err
+	}
+	podSpec := pod.Spec
+	return &podSpec, nil
+}
+
+func (p *Pod) SetImages(ctx context.Context, path string, containersPatch map[string]string, initContainersPatch map[string]string) error {
+	ns, n := client.Namespaced(path)
+	auth, err := p.Client().CanI(ns, "v1/pod", []string{client.PatchVerb})
+	if err != nil {
+		return err
+	}
+	if !auth {
+		return fmt.Errorf("user is not authorized to patch a deployment")
+	}
+	if manager, isManaged, err := p.isControlled(path); isManaged {
+		if err != nil {
+			return err
+		}
+		return fmt.Errorf("Unable to set image. This pod is managed by %s. Please set the image on the controller", *manager)
+	}
+	jsonPatch, err := GetJsonPatch(containersPatch, initContainersPatch)
+	if err != nil {
+		return err
+	}
+	dial, err := p.Client().Dial()
+	if err != nil {
+		return err
+	}
+	_, err = dial.CoreV1().Pods(ns).Patch(
+		ctx,
+		n,
+		types.StrategicMergePatchType,
+		[]byte(jsonPatch),
+		metav1.PatchOptions{},
+	)
+	return err
+}
+
+func (p *Pod) isControlled(path string) (*string, bool, error) {
+	pod, err := p.GetInstance(path)
+	if err != nil {
+		return nil, false, err
+	}
+	references := pod.GetObjectMeta().GetOwnerReferences()
+	if len(references) != 0 && *references[0].Controller == true {
+		return getController(references[0]), true, nil
+	}
+	return nil, false, nil
+}
+
+func getController(reference metav1.OwnerReference) *string {
```

Comment:
Bad form just return string and just return fmt.Sprintf("%s/%s", reference.Kind, reference.Name)